src/test/scala/uk/gov/ons/bi/dataload/ubrn/LinksPreprocessorFlatSpec.scala

```diff
@@ -0,0 +1,68 @@
+package uk.gov.ons.bi.dataload.ubrn
+
+import org.scalatest.{FlatSpec, Matchers}
+import uk.gov.ons.bi.dataload.model.BiSparkDataFrames
+import uk.gov.ons.bi.dataload.SparkCreator
+import uk.gov.ons.bi.dataload.helper.DataframeAsserter
+import org.apache.spark.sql.Row
+
+class LinksPreprocessorFlatSpec extends FlatSpec with Matchers with SparkCreator with DataframeAsserter {
+
+  import spark.implicits._
+
+  val lpp = new LinksPreprocessor(ctxMgr)
+
+  "Links Preprocessor" should "Read set previous Links to empty Dataframe if error" in {
```

Comment:
ScalaTest conventions is `"subject" should "expected behaviour"`. So in this case we should have:

`"readPrevLinks" should "return an empty DataFrame when given invalid inputs"`